obj/objects.cpp

```diff
@@ -1,315 +1,266 @@
+  else if (time1 > constants::kAccuracy && time2 > constants::kAccuracy) {
+    float time = std::min(time1, time2);
+    core::Vec3 collision = ray.at(time);
+    *hitData = core::RayHitData(time, normal(collision), ray, freq);
+    return true;
+  }
+  return false;
+}
+
+std::ostream &operator<<(std::ostream &os, const Sphere &sp) {
+  return os << "Sphere origin: " << sp.getOrigin()
+            << ", radius: " << sp.getRadius() << " [m]";
+}
+
+// GETTERS AND SETTERS
+void Sphere::setRadius(float rad) { radius_ = rad; }
+float Sphere::getRadius() const { return radius_; }
+
+std::ostream &operator<<(std::ostream &os, const SphereWall &sp) {
+  return os << "SphereWall origin: " << sp.getOrigin()
+            << ", radius: " << sp.getRadius() << " [m]";
+}
+
+// OPERATORS
+EnergyCollector &EnergyCollector::operator=(const EnergyCollector &other) {
+  if (other == *this) {
+    return *this;
+  }
+
+  this->setOrigin(other.getOrigin());
+  this->setRadius(other.getRadius());
+  energy_ = other.getEnergy();
+
+  return *this;
+}
+
+std::ostream &operator<<(std::ostream &os, const EnergyCollector &collector) {
+  return os << "Energy Collector. Origin: " << collector.getOrigin()
+            << ", Radius: " << collector.getRadius();
+}
+
+bool operator==(const EnergyCollector &left, const EnergyCollector &right) {
+  return (left.getOrigin() == right.getOrigin() &&
+          left.getRadius() == right.getRadius() &&
+          left.getEnergy() == right.getEnergy());
+}
+
+// METHODS
+
+float EnergyCollector::distanceAt(const core::Vec3 &positionHit) const {
+  return (getOrigin() - positionHit).magnitude();
+}
+
+void EnergyCollector::collectEnergy(const core::RayHitData &hitdata) {
+  // TODO: Energy distribution between many collectors
+  energy_ += hitdata.energy();
+}
+
+// GETTERS AND SETTERS
+void EnergyCollector::setEnergy(float en) { energy_ = en; }
+float EnergyCollector::getEnergy() const { return energy_; }
+void EnergyCollector::addEnergy(float en) { energy_ += en; }
+
+// CONSTRUCTORS
+
+TriangleObj::TriangleObj(const core::Vec3 &point1, const core::Vec3 &point2,
+                         const core::Vec3 &point3)
+    : point1_(point1), point2_(point2), point3_(point3) {
+  this->arePointsValid();
+  this->refreshAttributes();
+}
+
+TriangleObj::TriangleObj(const TriangleObj &other) {
+  this->setPoint1(other.point1());
+  this->setPoint2(other.point2());
+  this->setPoint3(other.point3());
+  this->refreshAttributes();
+}
+
+// OPERATORS
+
+TriangleObj &TriangleObj::operator=(const TriangleObj &other) {
+  this->setPoint1(other.point1());
+  this->setPoint2(other.point2());
+  this->setPoint3(other.point3());
+
+  this->refreshAttributes();
+
+  return *this;
+}
+
+bool operator==(const TriangleObj &left, const TriangleObj &right) {
+  // if other triangle has the same points but declared in different order,
+  // they will be still equal.
+  std::vector<core::Vec3> vertexVec(
```

Comment:
I can. I will do it tommorow... I did a friend because that what I read in the book as a recomendation ;)

Reply:
Not sure I get it, where is this `friend` you made?

Also, what recommendation? Remember what I said about explaining things?

Reply:
![image](https://user-images.githubusercontent.com/45213495/97079189-2f9cb480-15f2-11eb-96b5-4749d4f85e40.png)
 its here in object.h ... yeah, I know. when you said it, I had already implemented it and I didnt changed it yet. There are way more cases with operator == because I implemented this operator everywhere with friend

Reply:
But how does it make my comment about moving the code creating vector of points to the TriangleClass invalid? O_o

Reply:
I misunderstood. I thought your comment is about operator==, because thats only what I can see here ;/
![image](https://user-images.githubusercontent.com/45213495/97308531-40069680-1861-11eb-9981-004fcb636854.png)

Reply:
and also this is werid because I used new method getPoints() whoch is not displayed here